packages/dev-approuter/lib/helpers.js

```diff
@@ -0,0 +1,133 @@
+const path = require("path");
+
+/**
+ * Parses the approuter configuration from an `xs-dev.json` file.
+ * If that file doesn't exist, it looks for an `xs-app.json` file.
+ * Also scans the routes of the config file and checks for the `dependency` property
+ * and stores those in `config.dependencyRoutes`, which is where the dev approuter will do its magic.
+ * @returns {Object} the approuter configuration including all `dependencyRoutes`.
+ */
+const parseConfig = () => {
+    let config;
+    let configFile;
+    try {
+        config = require(path.join(process.cwd(), "xs-dev.json"));
+        configFile = "xs-dev.json";
+    } catch {
+        config = require(path.join(process.cwd(), "xs-app.json"));
+        configFile = "xs-app.json"
+    }
+    config.dependencyRoutes = {};
+    config.routes?.forEach(route => {
+        if (route.dependency) {
+            if (config.dependencyRoutes[`${route.dependency}`]) {
+                throw new Error(`Duplicate dependency "${route.dependency}" found in file ${path.join(process.cwd(), configFile)}.`);
+            } else {
+                config.dependencyRoutes[`${route.dependency}`] = route;
+            }
+        }
+    });
+    return config;
+};
+
+/**
+ * Applies the `config.dependencyRoutes` to the `config.routes` and removes them.
+ * @param {Object} config - the approuter configuration including all `dependencyRoutes`.
+ * @returns {Object} config - the approuter configuration that can be used to start the approuter.
+ */
+const applyDependencyConfig = (config) => {
+    config.routes?.forEach(route => {
+        if (route.dependency) {
+            route = config.dependencyRoutes[route.dependency];
+        }
+    });
+    delete config.dependencyRoutes;
+    return config;
+};
+
+/**
+ * Adds a destination to `process.env.destinations` for a given module.
+ * If a destination with this `name === moduleId` already exists, no new destination will be created.
+ * @param {String} moduleId - the id of the module that a destination is created for.
+ * @param {Number} port - the port of the localhost that the destination should point to.
+ * @param {String} mountPath - the path the module was mounted to and the destination should point to.
+ */
+const addDestination = (moduleId, port, mountPath) => {
+    let destinations = [];
+    if (process.env.destinations) {
+        destinations = JSON.parse(process.env.destinations);
+    }
+
+    let url;
+    if (mountPath) {
+        url = `http://localhost:${process.env.PORT || 5000}${mountPath}`;
+    } else {
+        url = `http://localhost:${port}`;
+    }
+    
+    // only add new destination if it's not already provided
+    let destinationAlreadyExists;
+    destinations.forEach(destination => {
+        const lowerCaseDestination = {};
+        Object.keys(destination).forEach(key => {
```

Comment:
You can use `Array.find` here to get the reference to the name property.

Reply:
We could simply test for `destination.name === moduleId`, but the tricky thing is that the keys of a destination are not case sensitive, meaning a `{"NaMe": "myDestination", ...}` is a valid thing as far as the approuter is concerned.

Reply:
I know, but in Array.find you can implement the function to get the value, e.g.:

```js
Object.entries({ "KeY": "value", "otHerKey": "otherValue" }).filter((entry) => entry[0].toLowerCase() === "otherkey")[0]?.[1] => returns "otherValue"
```

Reply:
Ok, but I think the current solution with `Array.some()` is already pretty neat.

Reply:
All good... Personal freedom... 😁